devcnairobi/devc-nairobi-bot
# feat: add getting started button.

Hi @ProfNandaa ,
I came across this project and was really interested in it. I saw your issues (#27 ) and I managed to add a getting started button on the bot.

**How to test**
Make sure that the firebase read rules are allowed.

Then you do a `POST` to `https://graph.facebook.com/v2.6/me/thread_settings?access_token=YOUR_PAGE_TOKEN` with this kind of payload to get the button on the bot:
```
{
  "setting_type":"call_to_actions",
  "thread_state":"new_thread",
  "call_to_actions":[
    {
      "payload":"USER_DEFINED_PAYLOAD"
    }
  ]
}
```

Then delete the conversation with your `development` bot so that when you initialize a new one, the button will show up.

I noticed there were no tests so I didn't put in any. Kindly check and would really like to hear your feedback.
